### DISCUSSION TITLE
PAG是否会导致输出视频成片的质量下降？

### DISCUSSION THREAD

**Discussion by weizmann** [Tencent/libpag#1660] [Q&A] [answered]
使用了一个单轨AE，直接导出了一个零特效的PAG文件。想测试下PAG是否会对画质有影响。

原视频ffmpeg拆帧到png，逐帧replace的时候，生成bmp图。

发现用PAG渲染的结果会比原视频的色彩差很多。想问下PAG是否会对画质效果有所损失？这种问题的排查思路是什么？

图片效果（注意看蘑菇的颜色、树木阴影的颜色）
![image](https://github.com/Tencent/libpag/assets/2306111/3b64e108-b654-4aff-bd0a-edae3cfbb462)

pag文件
[empty.pag.zip](https://github.com/Tencent/libpag/files/12337352/empty.pag.zip)

**Comment by kevingpqi123** [comment, marked as answer]
测试的思路和方向都是有问题的：
首先对比的来源不应该是视频，视频的清晰度和视频的码率有关，ffmpeg 拆分 png 也可能存在画质损失，pag 渲染导出视频也存在编码码率的问题，还有可能导致画质损失

应该对比的是啥：
在 AE 中设计矢量的动画，导出 pag，在桌面端预览效果，看与 AE 的渲染是否存在差异， 这可以判断渲染层面与数据源是否存在偏差
至于导出视频成片，更多取决于编码视频的参数，如码率等
